tests/test_update_query.py

```diff
@@ -117,6 +117,34 @@ def test_with_query_list_int():
     assert str(url.with_query([("a", 1)])) == "http://example.com/?a=1"
 
 
+@pytest.mark.parametrize(
+    "query,expected",
+    [
+        pytest.param({"a": [1, 2]}, "?a=1&a=2", id="list"),
+        pytest.param({"a": (1, 2)}, "?a=1&a=2", id="tuple"),
+        pytest.param({"a": ["1", 2]}, "?a=1&a=2", id="mixed types"),
+        pytest.param({"a": 1, "b": [2, 3]}, "?a=1&b=2&b=3", id="single then list",),
+        pytest.param({"a": [1, 2], "b": 3}, "?a=1&a=2&b=3", id="list then single",),
```

Comment:
could also include potentionally buggy cases
```suggestion
        pytest.param({"a": [1, 2], "b": 3}, "?a=1&a=2&b=3", id="list then single"),
        pytest.param({"a": "1&a=2", "b": 3}, "?a=1%26a=2&b=3", id="ampersand value"),
        pytest.param({"a": ["1&a=2", 3], "b": 3}, "?a=1%26a=2&a=3&b=3", id="ampersand value mixed with a list"),
```

Reply:
I think there's no need to test for `"1&a=2"`, because this test is about the use of sequences, so the latter should be enough. We already cover a similar case in `test_with_query_multidict_with_unsafe` and `test_with_query_params`.

Reply:
I'd still have it here. Because updates in one part of code can sometimes break others. Better safe than sorry.

Reply:
I will update the `test_with_query_params` and `test_with_query_params2` to use `parametrize` too and also add your new case. Does that sound good?